### PR TITLE
Fix NavigationRail Indicator alignment for `NavigationRailLabelType.none`

### LINKED ISSUE #97753: NavigationRailDestination icons are incorrectly aligned with labelType set to `NavigationRailLabelType.none`
## Steps to Reproduce

This was checked inside DartPad and the behaviour is the same so it doesn't require me to run  'flutter doctor' or any flutter commands; the bug is also quite simple.

1. Use the Stack widget inside the icon label in the NavigationRailDestionation in a NavigationRail Widget.
2. Ensure that the labeltype of NavigationRail is set to none (labelType: NavigationRailLabelType.none). See code below for more details.
2. The Stack Widget creates an unwanted spacing and doesn't behave as expected.
3. See the code below in dartpad for a visual demostration. Code is Commented.

**Expected results:** 
Run Dart Pad linked below.

**Actual results:**
Run Dart Pad linked below.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: const Scaffold(
        body: Center(
          child: TestScreenBody(),
        ),
      ),
    );
  }
}
class TestScreenBody extends StatelessWidget {
  const TestScreenBody({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Column(
      mainAxisSize: MainAxisSize.min,
      children: [
        const Spacer(
          flex: 1,
        ),
        Flexible(
          flex: 9,
          child: Row(
            mainAxisSize: MainAxisSize.min,
            children: [
              Flexible(
                flex: 9,
                child: Padding(
                  padding: const EdgeInsets.all(15),
                  /*
                  We see the Stack Widget behaving properly when using inside a generic 'icon:'
                  In this example an IconButton.
                  */
                  child: IconButton(
                    onPressed: null,
                    icon: Stack(
                      children: const [
                        Icon(Icons.umbrella),
                        Positioned(
                          top: 0,
                          right: 0,
                          child: Text(
                            'Test',
                            style: TextStyle(fontSize: 10, color: Colors.red),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ),
              Flexible(
                flex: 3,
                child: NavigationRail(
                    labelType: NavigationRailLabelType.none, /* <-------------- ONLY HAPPENS WHEN LABEL TYPE IS NONE */
                    destinations: [
                      NavigationRailDestination(
                        /*
                  We are inside the NavigationRail.
                  We see the Stack Widget behaving in an unwanted way inside the 'icon:', 
                  there exist some unwanted spacing.
                  */
                        icon: Stack(
                          children: const [
                            Icon(Icons.umbrella),
                            Positioned(
                              top: 0,
                              right: 0,
                              child: Text(
                                'Test',
                                style:
                                    TextStyle(fontSize: 10, color: Colors.red),
                              ),
                            ),
                          ],
                        ),
                        label: const Text('Bugged'),
                      ),
                      const NavigationRailDestination(
                        /*
                   We are inside the NavigationRail.
                   Normal Icon given to the 'icon:' widget behaves as expected.
                  */
                        icon: Icon(Icons.umbrella),
                        label: Text('Umbrella'),
                      ),
                      NavigationRailDestination(
                        /*
                  We are inside the NavigationRail.
                  Temporary Fix is to wrap it inside a Center Widget
                  */
                        icon: Center(
                          child: Stack(
                            children: const [
                              Icon(Icons.umbrella),
                              Positioned(
                                top: 0,
                                right: 0,
                                child: Text(
                                  'Test',
                                  style: TextStyle(
                                      fontSize: 10, color: Colors.red),
                                ),
                              ),
                            ],
                          ),
                        ),
                        label: const Text('Bug'),
                      ),
                    ],
                    selectedIndex: 0),
              ),
            ],
          ),
        ),
      ],
    );
  }
}
```
</details>
<details>
<summary>DartPad Link</summary>

[https://dartpad.dev/?id=d1766c74b8bfe2bd81d09905063845a3](https://dartpad.dev/?id=d1766c74b8bfe2bd81d09905063845a3)

</details>

**Temporary Solution**
Wrap the Stack Widget inside a Center Widget.

### PR DESCRIPTION
**Pull request by TahaTesser**
Details: https://github.com/flutter/flutter/issues/97753#issuecomment-1032552272

fixes https://github.com/flutter/flutter/issues/97753

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/navigation_rail_test.dart [hunk 1]
```diff
@@ -2349,6 +2349,50 @@ void main() {
     expect(indicator.width, 56);
     expect(indicator.height, 32);
   });
+
+  testWidgets('NavigationRailDestination has center aligned indicator - [labelType]=none', (WidgetTester tester) async {
+      // This is a regression test for
+      // https://github.com/flutter/flutter/issues/97753
```
Comment (johnsonmh) [inline packages/flutter/test/material/navigation_rail_test.dart:2355]: Formatting nit: I think this comment should be back two spaces, to line up with the `await _pumpNavigationRail` below.

**Review by johnsonmh — APPROVED** [review]
LGTM thanks for the fix

File: packages/flutter/test/material/navigation_rail_test.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/test/material/navigation_rail_test.dart]: My pleasure! That's weird, I might have accidentally tabbed before commit, thanks

File: packages/flutter/lib/src/material/navigation_rail.dart [hunk 2]
```diff
@@ -579,12 +579,14 @@ class _RailDestination extends StatelessWidget {
         final Widget iconPart = SizedBox(
           width: minWidth,
           height: minWidth,
-          child: _AddIndicator(
-            addIndicator: useIndicator,
-            indicatorColor: indicatorColor,
-            isCircular: true,
-            indicatorAnimation: destinationAnimation,
-            child: themedIcon,
+          child: Align(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_rail.dart:582]: If this is the same as `Center`, Center would be a little clearer.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**